File: netty/src/main/java/cn/elmi/sample/netty/chat/ChatClientHandler.java
package cn.elmi.sample.netty.chat;

import io.netty.channel.ChannelHandlerContext;
import io.netty.channel.SimpleChannelInboundHandler;

/**
 * @version 1.0
 * @authur Arthur
 */
public class ChatClientHandler extends SimpleChannelInboundHandler<String> {

    @Override
    protected void channelRead0(ChannelHandlerContext ctx, String msg) throws Exception {
        System.out.println(msg);
    }

}

File: netty/src/main/kotlin/cn/elmi/sample/netty/chat/ChatClientHandler.kt
package cn.elmi.sample.netty.chat

import io.netty.channel.ChannelHandlerContext
import io.netty.channel.SimpleChannelInboundHandler

/**
 * @version 1.0
 * @authur Arthur
 */
class ChatClientHandler : SimpleChannelInboundHandler<String?>() {
    @Throws(Exception::class)
    override fun channelRead0(ctx: ChannelHandlerContext, msg: String?) {
        println(msg)
    }
}